pkg/webhook/controlplaneexposure/ensurer.go

```diff
@@ -80,7 +80,6 @@ func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, gctx gconte
 
 	if c := extensionswebhook.ContainerWithName(new.Spec.Template.Spec.Containers, "kube-apiserver"); c != nil {
 		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--advertise-address=", address)
-		c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--external-hostname=", address)
```

Comment:
@ialidzhikov, @ScheererJ Do you have an idea on the impact for the SNI case?

ref: https://github.com/gardener/gardener-extension-provider-openstack/issues/366#issuecomment-996698944

Reply:
@dkistner As mentioned by @ialidzhikov in #366, this should have no impact whatsoever on the SNI case as the logic jumps out of the function in line 63 due to `v1beta1helper.IsAPIServerExposureManaged(new)` returning `true` in case SNI is enabled.

I am not sure why the external hostname is overwritten here. As far as I can tell the external hostname is always set by gardener since 1.33. Maybe in the past, the certificates had the IPs as subject and not the DNS name.